app/fullService/api/assignAddressForAccount.ts

```diff
@@ -17,17 +17,18 @@ const assignAddressForAccount = async ({
   accountId,
   metadata,
 }: AssignAddressForAccountParams): Promise<AssignAddressForAccountResult> => {
-  const { result, error } = await axiosFullService(ASSIGN_ADDRESS_FOR_ACCOUNT_METHOD, {
-    accountId,
-    metadata,
-  });
+  const { result, error }: AxiosFullServiceResponse<AssignAddressForAccountResult> =
+    await axiosFullService(ASSIGN_ADDRESS_FOR_ACCOUNT_METHOD, {
+      accountId,
+      metadata,
+    });
 
   if (error) {
     // TODO - I'll write up a better error handler
     const errorMessage = error === 'Database(PasswordFailed)' ? 'Incorrect Password' : error;
     throw new Error(errorMessage);
   } else {
-    return result;
+    return result as AssignAddressForAccountResult;
```

Comment:
given that both `result` and `error` are optional here, we could avoid the cast and guard against the case of no result by changing L26 to `if (error || !result) {`

Reply:
Thank you! I was totally blanking on the right way to handle this. Will update throughout